### DISCUSSION TITLE
How do I get a building address?

### DISCUSSION THREAD

**Discussion by pelepelin** [osmandapp/OsmAnd#11925] [Q&A]
I've just installed OsmAnd~ 3.9.10 from F-Droid, downloaded suggested local map (Ru, Saint-Petersburg, 2021-06-01) and looking around. I can't find such a basic function like viewing full address of a building. Each building around my place has a number visible on the map. On the osm.org I see that each building has a full address like this https://www.openstreetmap.org/way/136170895

tag | val
-- | --
addr:city | Санкт-Петербург
addr:country | RU
addr:housenumber | 69
addr:postcode | 195427
addr:street | Светлановский проспект

However, if I long-press inside of buildings in OsmAnd, on some buildings I get their addresses, on some buildings I get the name of a nearby street or just the city and district (like on the one above).

If I tap the number on the building, I get that number back in the panel, but the location in small font below is just something nearby like "..., аллея Глушко" in this case.

What am I missing here?

**Comment by thompehw** [comment]
I can replicate  your example: clicking inside [https://www.openstreetmap.org/way/136170895](https://www.openstreetmap.org/way/136170895)  doesn't show the housenumber in the address. 

I thought it might be a confusion with the stairway entrance nodes on this building having their own addr:flats tags. I experimented with similar buildings nearby and noticed that long-pressing inside [https://www.openstreetmap.org/way/22995740](https://www.openstreetmap.org/way/22995740)  does give this building's housenumber if I click in its centre, but towards either end I get the housenumber of the adjacent buildings instead.

I hope this might stimulate someone who knows the explanation! Best wishes.

**Comment by vshcherb** [comment]
Unfortunately OsmAnd doesn't detect address correctly for large buildings cause it stores only 1 point from the building and then find closest point. 

![Screen Shot 2021-06-18 at 18 59 48](https://user-images.githubusercontent.com/1042025/122594560-73539b00-d067-11eb-8c40-b4a0aeda9e07.png)

**Reply by pelepelin** [reply]
Any chance it can be fixed?
At least the second case, where the number is explicitly tapped, should be simple enough. When the number is tapped, the building is highlighted, so they are linked correctly. However, the popup panel shows only the number in the first bold line and not the full address and the second line still contains nearby address.

**Reply by vshcherb** [reply]
I hope so
